packages/daemon/src/service/trace-manager.ts

```diff
@@ -0,0 +1,250 @@
+import { Transform, TransformCallback } from 'stream';
+import { EventEmitter } from 'events';
+import { open, close, write } from 'fs-extra';
+import { provide, scope, ScopeEnum } from 'midway';
+import { StringDecoder } from 'string_decoder';
+import { generateId, PipelineStatus, PluginTypeI } from '@pipcook/pipcook-core';
+import Debug from 'debug';
+
+const debug = Debug('tracer');
+
+export type PipcookEventType = 'log' | 'job_status';
+
+/**
+ * base pipcook event, defined the fields `type` and `data`
+ */
+export class BasePipcookEvent {
+  data?: any;
+  constructor(public type: PipcookEventType) {}
+}
+
+/**
+ * pipcook event data type for job status change
+ */
+export class JobStatusChangeEvent extends BasePipcookEvent {
+  data: {
+    jobStatus: PipelineStatus;
+    step?: PluginTypeI;
+    stepAction?: 'start' | 'end';
+    queueLength?: number;
+  };
+  constructor(jobStatus: PipelineStatus, step?: PluginTypeI, stepAction?: 'start' | 'end', queueLength?: number) {
+    super('job_status');
+    this.data = { jobStatus, step, stepAction, queueLength };
+  }
+}
+
+export type LogLevel = 'info' | 'warn' | 'error';
```

Comment:
```suggestion
type LogLevel = 'info' | 'warn' | 'error';
```

It's used in this file only.